php-classes/Slate/CBL/Demonstrations/StudentDashboardRequestHandler.php

```diff
@@ -122,10 +122,21 @@ public static function handleCompletionsRequest()
         $completions = [];
 
         foreach ($competencies AS $Competency) {
-            $completions[] = array_merge([
-                'StudentID' => $Student->ID,
-                'CompetencyID' => $Competency->ID
-            ], $Competency->getCompletionForStudent($Student));
+            $StudentCompetency = StudentCompetency::getCurrentForStudent($Competency, $Student);
+
+            if ($StudentCompetency) {
+                $completions[] = $StudentCompetency->getCompletion();
+            } else {
+                $completions[] = [
```

Comment:
@nbey let's create a method `StudentCompetency::getBlankCompletion($Student, $Competency)` for returning this data structure and use it in the teacher request handler too so that as we add to / evolve this data structure there are only two places it is defined and they're both in the same class